Java/core/src/main/java/com/amazon/randomcutforest/store/INodeStore.java

```diff
@@ -182,14 +209,12 @@
      */
     public int getSibling(int parent, int node);
 
-    /**
-     * the following returns a map of (leaves,correponding parent) pairs to enable
-     * validation in serialization
-     *
-     * @return map of leaf,corresponding pairs in use for the node store
-     */
-    default Map<Integer, Integer> getLeavesAndParents() {
-        return null;
-    }
+    boolean isLeaf(int index);
+
+    int getCapacity();
+
+    int computeLeafIndex(int index);
```

Comment:
Does this really need to be in the interface? It seems like this should be handled internally in the node store.